### PR TITLE
Add C++ Solution + Editorial to Mahmoud and Ehab and the Function (Silver)

### PR DESCRIPTION
**Pull request by ryanchou-dev**
im back -w-

---
_If any of the below doesn't apply to this Pull Request, mark the checkbox as done._

- [x] I have tested my code.
- [x] I have added my solution according to the steps [here](https://usaco.guide/general/adding-solution#steps).
- [x] I have followed the code conventions mentioned [here](https://usaco.guide/general/adding-solution/#code-conventions), which includes the following: 
  - I understand that if it is clear that I have not attempted to follow these guidelines (ex. if I have not used tabs to indent), my PR will be closed.
  - If changes are requested, I will re-request the review after making them.

### CODE REVIEW

File: solutions/cf-862E.mdx [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+**Time Complexity:** $\mathcal{O}(N + (M + Q)\log M)$
+
+<LanguageSection>
+<CPPSection>
+
+```cpp
+#include <iostream>
+#include <vector>
+#include <algorithm>
+using std::vector;
```
Comment (jessechoe10) [inline solutions/cf-862E.mdx:37]: pls use ```using namespace std;```

**Review by jessechoe10 — APPROVED** [review]

File: solutions/cf-862E.mdx [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+	}
+	for (int i = 0; i < m; i++) {
+		std::cin >> b[i];
+	}
+
+	/*
+	 * calculate the odd and even sums, 
+	 * we care about their sums in the case 
+	 * where we start on a index with a different parity
+   */
```
Comment (SansPapyrus683) [inline solutions/cf-862E.mdx:56]: indentation sus

File: solutions/cf-862E.mdx [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+	}
+
+	std::sort(seg_b.begin(), seg_b.end());
+	
+	long long sum_a = 0;
+	for (int i = 0; i < n; i++) {
+		sum_a += (i & 1 ? -a[i] : a[i]);
+	}
+
+	auto qry = [&](auto it) {
```
Comment (SansPapyrus683) [inline solutions/cf-862E.mdx:94]: why put this in a function?

File: solutions/cf-862E.mdx [hunk 4]
```diff
@@ -0,0 +1,138 @@
...
+#include <iostream>
+#include <vector>
+#include <algorithm>
+using std::vector;
+
+int main() {
+	int n, m, q;
+	std::cin >> n >> m >> q;
+	
+	vector<int> a(n), b(m);
```
Comment (SansPapyrus683) [inline solutions/cf-862E.mdx:43]: declarations on sep. lines

File: solutions/cf-862E.mdx [hunk 5]
```diff
@@ -0,0 +1,138 @@
...
+		std::cin >> a[i];
+	}
+	for (int i = 0; i < m; i++) {
+		std::cin >> b[i];
+	}
+
+	/*
+	 * calculate the odd and even sums, 
+	 * we care about their sums in the case 
+	 * where we start on a index with a different parity
```
Comment (SansPapyrus683) [inline solutions/cf-862E.mdx:55]: i'd like if you could be a tad bit more formal in your comments
i'm fine with this but it just contradicts with the mood of the editorial

**Review by SansPapyrus683 — CHANGES_REQUESTED** [review]

File: solutions/cf-862E.mdx (hunk 2 above)
Comment (ryanchou-dev) [reply solutions/cf-862E.mdx:56]: oop

File: solutions/cf-862E.mdx (hunk 3 above)
Comment (ryanchou-dev) [reply solutions/cf-862E.mdx:94]: it's neater?

Comment (SansPapyrus683) [reply solutions/cf-862E.mdx:97]: fair enough

File: solutions/cf-862E.mdx [hunk 6]
```diff
@@ -0,0 +1,141 @@
...
+
+			if (loc != seg_b.begin()) {
+				ans = std::min(ans, llabs(*(std::prev(loc)) - sum_a));
+			}
+
+			std::cout << ans << std::endl;
+		}
+	};
+
+	qry(std::lower_bound(seg_b.begin(), seg_b.end(), sum_a));
```
Comment (SansPapyrus683) [inline solutions/cf-862E.mdx:117]: wait why do a `qry` @ the start?

**Review by SansPapyrus683 — APPROVED** [review]

File: solutions/cf-862E.mdx (hunk 6 above)
Comment (ryanchou-dev) [reply solutions/cf-862E.mdx:117]: you have to answer it without any updates first

**Review by nwatx — APPROVED** [review]

**Comment by SansPapyrus683** [comment]
1 CA + two mods, should be good enough

**Pull request merged**